src/libmeasurement_kit/ooni/telegram.cpp

```diff
@@ -0,0 +1,194 @@
+// Part of measurement-kit <https://measurement-kit.github.io/>.
+// Measurement-kit is free software. See AUTHORS and LICENSE for more
+// information on the copying conditions.
+
+#include "../common/utils.hpp"
+#include "../ooni/constants.hpp"
+#include <measurement_kit/ooni.hpp>
+
+namespace mk {
+namespace ooni {
+
+using namespace mk::report;
+
+static void tcp_many(const std::vector<std::string> ip_ports,
+                     Var<Entry> entry,
+                     Callback<Error> cb,
+                     Var<Reactor> reactor,
+                     Var<Logger> logger) {
+    // C++20 should have continuations in std; until then,
+    // we copy the ghetto pattern from web_connectivity
```

Comment:
I think there is already an abstraction that can be used to do that in `common`. It is called `parallel` and runs a specific algorithm over a vector of input. I think that we should:

1) check whether we can refactor that code to work in this case (and also in web_connectivity's case)

2) check which should be the expected syntax of C++20 continuations and see whether we can modify the API of `parallel` such that, when we switch to C++20, the transition is as smooth as possible.

Regarding parallel, useful links are:

a) [documentation](https://github.com/measurement-kit/measurement-kit/blob/de0c4c4abef0ebeab81691015c7c92dec7696884/doc/api/common/parallel.md)

b) [example of usage](https://github.com/measurement-kit/measurement-kit/blob/1aa5bd817dfa309b8574e919526d8f7862060f53/src/libmeasurement_kit/nettests/runnable.cpp#L304)